src/shaders/disk_andrews2009_frag.glsl

```diff
@@ -47,14 +47,11 @@ void main(void)
 {
     // Compute the radius and height of this fragment
     float radius = sqrt(ModelPosition.x * ModelPosition.x + ModelPosition.y * ModelPosition.y);
-    float height = abs(ModelPosition.z);
+    float height = atan(abs(ModelPosition.z),radius);
```

Comment:
Hi Brian,

As both Fabien and both worked out that atan2 would be more appropriate, I am curious what 
he thinks about the performance issue.  Fabien, what are your thoughts?  If it isn't a big deal, I 
will bow to Brian's judgment on this.

Peace,
Rob

-- 
J. Robert Parks
Sponsored Funded Professional Astronomer
Department of Physics & Astronomy
Georgia State University
25 Park Place South
Room# 615
Atlanta, GA 30303
404-413-6081
On Wednesday, July 01, 2015 05:59:41 PM Brian Kloppenborg wrote:


In src/shaders/disk_andrews2009_frag.glsl[1]:
> @@ -47,14 +47,11 @@ void main(void)
>  {
>      // Compute the radius and height of this fragment
>      float radius = sqrt(ModelPosition.x * ModelPosition.x + ModelPosition.y * 
ModelPosition.y);
> -    float height = abs(ModelPosition.z);
> +    float height = atan(abs(ModelPosition.z),radius);
Humm. The use of atan2 in the shader will result in a very large performance loss. Lets rewrite 
Andrew's equation to use z instead (he has a lot of odd conventions if I recall correctly). We 
can just make note of this in the help text and on the SIMTOI wiki. 
—Reply to this email directly or view it on GitHub[2].[3]




--------
[1] https://github.com/bkloppenborg/simtoi/pull/133#discussion_r33739846
[2] https://github.com/bkloppenborg/simtoi/pull/133/files#r33739846
[3] 
https://github.com/notifications/beacon/AJgFFTbWhi5jHyudGBcD99S0Muum7tb_ks5oZIR9
gaJpZM4FQKMR.gif